app/routes/book_routes.py

```diff
@@ -49,6 +49,17 @@ def get_one_book(book_id):
         "description": book.description,
     }
 
+@books_bp.put("/<book_id>")
+def update_book(book_id):
+    book = validate_book(book_id)
+    request_body = request.get_json()
+
+    book.title = request_body.get("title")
+    book.description = request_body.get("description")
```

Comment:
This made me go back and check the POST route for comparison. PUT says to replace the resource, and so it should have the same data requirements as a POST. By using `get` here, if a key is missing, we'll get back `None`, which then fails in the `commit` due to the columns being not nullable (raises `NotNullViolation`).

As I suggested for the POST, I'd rather stick with direct key access here (possible with a callout in the text about there being the possibility of data error that the reader can either handle here or that we'll address in the future). In that case, we should also doa refactor for model updates that will catch the error the same way the POST does.